### PR TITLE
Fix KF Dex authentication bug

### LINKED ISSUE #1636: Handle Kubeflow authentication/authorization errors
**Describe the issue**
See below.

**To Reproduce**
1. Provision a _secured_ KFP deployment 
1. Create a runtime configuration in Elyra that meets one of the following criteria
    - no user name + password specified
    - invalid user name specified
    - invalid password specified
2. Create a notebook
3. Create a pipeline
4. Add notebook to pipeline
4. Using the "invalid" runtime configuration run pipeline OR export pipeline DSL and run DSL
5. Observe failure 
   ```
   f"pipeline submitted: {api_endpoint}/#/runs/details/{run.id}",
    AttributeError: 'NoneType' object has no attribute 'id'
   ```
During a preliminary investigation I've noticed that there are several silent errors that are never detected. For example, experiment creation fails without raising an error. It appears that the `kfp` client (tested with `0.7.0`) doesn't seem to detect/handle/expose authentication/authorization issues caused by missing/invalid credentials.

**Expected behavior**
Elyra analyzes the KFP response for signs that an API call might have failed even though no exception is raised or error returned.

**Deployment information**
Describe what you've deployed and how:
 - Elyra version: 2.3/master (likely also in earlier versions)
 - Kubeflow 1.3 with security enabled (e.g. via DEX) - didn't test earlier versions

### PR DESCRIPTION
**Pull request by ptitzler**
This PR 
 - handles Kubeflow [Dex] authentication issues
    - missing credentials
    - partial credentials
    - invalid credentials
 - closes #1636 

### What changes were proposed in this pull request?
- Fix faulty code
- Add log message for problem determination, should authentication fail
   ```
    Kubeflow authentication info: {'endpoint': 'http://custard1.fyre.ibm.com:31380', 'endpoint_response_url': 'http://custard1.fyre.ibm.com:31380/dex/auth/local?req=ywzv...cy', 'endpoint_secured': True, 'authservice_session_cookie': None}
   ```
- Add proper error message

  image

### How was this pull request tested?

- [x] connect to unsecured KF without credentials in the RTC (expected to succeed)
- [x] connect to unsecured KF with credentials in the RTC (expected to succeed)
- [x] connect to unsecured KF with only user name defined in the RTC (expected to succeed)
- [x] connect to unsecured KF with only password defined in the RTC (expected to succeed)
- [x] connect to secured KF without credentials in the RTC (expected to fail)
- [x] connect to secured KF with invalid credentials in the RTC (expected to fail)
- [x] connect to secured KF with only user name defined in the RTC (expected to fail)
- [x] connect to secured KF with only password defined in the RTC (expected to fail)
- [x] connect to secured KF with valid credentials in the RTC (expected to succeed)

Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by ptitzler** [comment]
Testing ^^ is still in progress

File: elyra/pipeline/processor_kfp.py [hunk 1]
```diff
@@ -77,23 +77,37 @@ def process(self, pipeline):
...
+            # and if it is try to authenticate with the user-provided credentials
+            # (if any were defined in the runtime configuration)
+
+            endpoint = api_endpoint.replace('/pipeline', '')
+            auth_info = \
+                KfpPipelineProcessor._get_user_auth_session_cookie(endpoint,
+                                                                   api_username,
+                                                                   api_password)
+
+            self.log.info(f"Kubeflow endpoint '{api_endpoint}' secured: "
```
Comment (kevin-bates) [inline elyra/pipeline/processor_kfp.py:90]: This feels more like a debug statement to me.

**Review by kevin-bates — APPROVED** [review]
Thanks @ptitzler - this looks good and I'm approving from a code perspective.  I can't test this, but trust you'll be doing that.  My only comment was a nit regarding the log level that feels more troubleshooting-related than informational.

File: elyra/pipeline/processor_kfp.py (hunk 1 above)
Comment (ptitzler) [reply elyra/pipeline/processor_kfp.py]: I changed the log level to debug and dumped the entire data structure.

**Comment by ptitzler** [comment]
All tests completed as expected, including informal test of debug output.

**Pull request merged**